R/server.R

```diff
@@ -968,31 +968,26 @@ server <- function(input, output, session) {
 
     df <- check_balance(selection(), params()$id, params()$year)
 
-    malmquist <- productivity::malm(
-      data = df$data, id.var = params()$id, time.var = params()$year,
-      x.vars = params()$inputs, y.vars = params()$outputs,
-      rts = input$malm_rts, orientation = input$malm_orientation, scaled = TRUE)
-
-    d <- malmquist$Changes[, c(1:6)]
-
-    colnames(d) <- c(params()$id, 'Year', 'Ref. year', 'Malmquist', 'Eff. change', 'Tech. change')
+    d <- compute_malmquist(
+      df$data, id = params()$id, time = params()$year,
+      input = params()$inputs, output = params()$outputs,
+      orientation = input$malm_orientation)
+    d  <- as.data.frame(d)[1:10]
 
     d
 
   })
 
-  malm.calc <- reactive({
-
-    d <- malm.mod()
-    d[, 3:6] <- sapply(d[,3:6], function(c) round(c, input$malm_round))
-    return(d)
-
-  })
-
   output$malm.render <- renderReactable({
     req(params()$year)
+    mlm_cols <- c('DMU', 'Time', 'Malmquist', 'Eff. change', 'Tech. change',
+                  'Input bias tech. change', 'Output bias tech. change',
+                  'Magnitude component', 'Pure eff. change',
+                  'Scale eff. change')
```

Comment:
We might want to use the `columns` argument in the `reachable()` function instead. However, this could also be something we do in a separate issue, as it would be much nicer with a utility function that sets default parameters for columns that can exist in multiple tables on the front end. Here is an example of such a utility function:

```r
default_columns <- function(columns, column_definitions = list()) {
  defaults <- list(
    client = colDef(name = 'Klient', maxWidth = 100),
    comp_reg_no = colDef(name = 'Org.nr', maxWidth = 100),
    voucher_no = colDef(name = 'Bilag')
  )
  column_list <- defaults[columns[columns %in% names(defaults)]]
  return(modifyList(column_list, column_definitions))
}
```

Reply:
I like this idea, but lets keep such a function for a separate PR.

Reply:
I agree that this could be a separate PR. I'll create a new issue.